EventFilter/RawDataCollector/src/RawDataMapperByLabel.cc

```diff
@@ -0,0 +1,117 @@
+
+class RawDataMapperByLabel: public edm::stream::EDProducer<> {
+public:
+    
+    ///Constructor
+    RawDataMapperByLabel(const edm::ParameterSet& pset);
+    
+    ///Destructor
+    ~RawDataMapperByLabel() override;
+    
+    void produce(edm::Event & e, const edm::EventSetup& c) override; 
+    
+    static void fillDescriptions(edm::ConfigurationDescriptions &);     
+private:
+    
+    typedef std::vector<edm::InputTag>::const_iterator tag_iterator_t;
+    typedef std::vector<edm::EDGetTokenT<FEDRawDataCollection> >::const_iterator tok_iterator_t;
+
+    std::vector<edm::InputTag> inputTags_ ;
+    std::vector<edm::EDGetTokenT<FEDRawDataCollection> > inputTokens_;
+    
+    edm::InputTag mainCollectionTag_ ;
+    edm::InputTag filledCollectionName_;
+    bool firstEvent_;
+    
+
+};
+
+
+RawDataMapperByLabel::RawDataMapperByLabel(const edm::ParameterSet& pset)
+ : inputTags_(pset.getParameter<std::vector<edm::InputTag>>("rawCollectionList")),
+   mainCollectionTag_(pset.getParameter<edm::InputTag>("mainCollection")),
+   filledCollectionName_(edm::InputTag("")),
+   firstEvent_(true)
+{
+
+  inputTokens_.reserve(inputTags_.size());
+  for(auto const& inputTag: inputTags_) {
+    inputTokens_.push_back(consumes<FEDRawDataCollection>(inputTag));
+  }
+
+  produces<FEDRawDataCollection>();
+   
+}
+
+
+RawDataMapperByLabel::~RawDataMapperByLabel(){
+
+}
+
+
+void RawDataMapperByLabel::produce(Event & e, const EventSetup& c){
+ 
+ bool alreadyACollectionFilled= false;
+ tag_iterator_t inputTag = inputTags_.begin();
+ for(tok_iterator_t inputTok = inputTokens_.begin(); inputTok != inputTokens_.end(); ++inputTok, ++inputTag  ) {
+   Handle<FEDRawDataCollection> input;
+   if(e.getByToken(*inputTok,input)){
+      if(input.isValid()){
+         if(alreadyACollectionFilled) throw cms::Exception("BadInput") << "Two input collections are present." << std::endl
+         << "Please make sure that the input dataset has only one FEDRawDataCollector collection filled";
+         
+         if(firstEvent_){  
+            filledCollectionName_ = *inputTag; 
+            alreadyACollectionFilled = true;
+            firstEvent_= false;  
+         }
+            
+         
+         if(!(filledCollectionName_==*inputTag)) throw cms::Exception("BadInput") << "The filled collection has changed!";
+            
+         if(!(mainCollectionTag_==filledCollectionName_)) e.put(std::make_unique<FEDRawDataCollection>(*input.product()));
+                      
+      }
+   }
+ }
+}
+
+void RawDataMapperByLabel::fillDescriptions(edm::ConfigurationDescriptions & descriptions) {
+    edm::ParameterSetDescription desc;
+    
+    std::vector<edm::InputTag> tmp_itvect;
+    tmp_itvect.push_back( edm::InputTag("rawDataCollector"));
+    tmp_itvect.push_back( edm::InputTag("rawDataRepacker"));
+    tmp_itvect.push_back( edm::InputTag("rawDataReducedFormat"));
+    
+    desc.add<std::vector<edm::InputTag>>("rawCollectionList",  tmp_itvect);
```

Comment:
(minor) 
IIUC ```desc.add<std::vector<edm::InputTag>>("rawCollectionList",  
   {"rawDataCollector", "rawDataRepacker", "rawDataReducedFormat"}) ``` should also work (perhaps with ```{{``` )

Reply:
done